### PR TITLE
Undo/Redo command

### PR DESCRIPTION
**Pull request by SpecOps2016**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/UndoCommand.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Select the given appointment and filter appointment list by given appointment.
+ */
+public class UndoCommand extends Command {
+    public static final CommandWord COMMAND_WORD =
+            new CommandWord("undo");
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Undoes the last command\n"
+            + "Example: " + COMMAND_WORD + " ";
```
Comment (guokweijie) [inline src/main/java/seedu/address/logic/commands/UndoCommand.java:17]: no need for whitespace after command word

File: src/main/java/seedu/address/logic/commands/RedoCommand.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Select the given appointment and filter appointment list by given appointment.
+ */
+public class RedoCommand extends Command {
+    public static final CommandWord COMMAND_WORD =
+            new CommandWord("redo");
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Redoes the last command\n"
+            + "Example: " + COMMAND_WORD + " ";
```
Comment (guokweijie) [inline src/main/java/seedu/address/logic/commands/RedoCommand.java:17]: no need for whitespace after command word

File: src/main/java/seedu/address/logic/commands/UndoCommand.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    public static final String MESSAGE_SUCCESS = "Undo successful";
+
+    public UndoCommand() {
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```
Comment (guokweijie) [inline src/main/java/seedu/address/logic/commands/UndoCommand.java:25]: can be more defensive and throw exceptions

File: src/main/java/seedu/address/model/Model.java [hunk 4]
```diff
@@ -199,4 +200,29 @@ public interface Model {
...
     void sortBills(Comparator<Bill> comparator, boolean isAscending);
+
+    /**
+     * Updates previous state of the address book.
+     */
+    void updateAddressBookHistory();
+
+    /**
+     * Updates state of the address book when undo command is executed.
+     */
```
Comment (guokweijie) [inline src/main/java/seedu/address/model/Model.java:211]: redo or undo command is executed?

**Review by guokweijie — COMMENTED** [review]
some minor fixes required

File: src/main/java/seedu/address/model/Model.java (hunk 4 above)
Comment (SpecOps2016) [reply src/main/java/seedu/address/model/Model.java:211]: when undo executed cos redo can only be done after undo is executed

**Pull request closed without merging**